jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryService.java

```diff
@@ -79,7 +80,7 @@
      * @return results mapped to objects from raw data set
      * @throws QueryNotFoundException in case there is no such query registered
      */
-    <T> T query(String queryName, QueryResultMapper<T> mapper, QueryContext queryContext, QueryParam...filterParams) throws QueryNotFoundException;
+    <T> T query(String queryName, QueryResultMapper<T> mapper, AdvancedQueryContext queryContext, QueryParam...filterParams) throws QueryNotFoundException;
```

Comment:
I'd prefer not to change the API here as the AdvancedQueryContext will only be used when there is a need to pass the sql like oder by clause which might not be so often. In all other cases it would be enough to rely on default QueryContext and by that we'll not break backward compatibility. Moreover this PR will have far less changes :)